File: sdl_generator/resources/fixed_array_fun_templates.erl
{{ErlName}}_to_bytelist(List) when length(List)=={{Desc}} ->
	[{{Type}}_to_bytelist || E<-List].

bytelist_to_{{ErlName}}(Bytelist) ->
	bytelist_to_{{ErlName}}(Bytelist, []).
bytelist_to_{{ErlName}}([], Result) ->
	lists:reverse(Result).
bytelist_to_{{ErlName}}(Bytelist, Result) ->
	{Elem, Rest} = parse_{{Type}}(Bytelist),
	bytelist_to_{{ErlName}}(Rest, [Elem|Result]).

parse_{{ErlName}}(Bytelist) ->
	parse_{{ErlName}}(Bytelist, [], 0).
parse_{{ErlName}}(Bytelist, Result, {{Desc}}) ->
	{lists:reverse(Result), Bytelist}.
parse_{{ErlName}}(Bytelist, Result, Cnt) ->
	{Elem, Rest} = parse_{{Type}}(Bytelist),
	parse_{{ErlName}}(Rest, [Elem|Result], Cnt+1).

File: sdl_generator/resources/fixed_array_fun_templates.c
byte * read_{{ErlName}}(byte *in, {{CName}} *result) {
	byte *current_in = in;

	for (int i=0; i<{{Desc}}; i++)
		current_in = read_{{Type}}(current_in, &(result[i]));

	return current_in;
}

byte * write_{{ErlName}}({{CName}} *array, byte *out, size_t *len) {
	byte *current_out = out;

	for (int i=0; i<{{Desc}}; i++)
		current_out = write_{{Type}}(&(array[i]), current_out, len);

	return current_out;
}

